app/controllers/topics_controller.rb

```diff
@@ -288,6 +288,18 @@ def update
       else
         return render_json_error(I18n.t('category.errors.not_found'))
       end
+
+      if category && topic_tags = params[:tags] || topic.tags.pluck(:name)
```

Comment:
I think we should add the paranthesis to ensure we get the priority right

```suggestion
      if category && topic_tags = (params[:tags] || topic.tags.pluck(:name))
```